src/uptime.rs

```diff
@@ -0,0 +1,115 @@
+use std::fs::read_to_string;
+
+use serde::{Deserialize, Serialize};
+
+use crate::Config;
+use crate::Module;
+
+const UPTIME_PATH: &str = "/proc/uptime";
+
+#[derive(Serialize, Deserialize)]
+#[serde(default)]
+pub struct Uptime {
+    header: String,
+
+    #[serde(skip)]
+    /// Uptime in seconds
+    uptime: u32,
+
+    /// When true, uptime will be outputted like: 2d 10h 3m
+    short_output: bool,
+}
+
+/// Returns system uptime in seconds
+fn get_uptime() -> u32 {
+    read_to_string(UPTIME_PATH)
+        .expect("Failed to read uptime file")
+        .split_once(' ')
+        .unwrap()
+        .0
+        .parse::<f32>()
+        .expect("Failed to parse uptime") as u32
+}
+
+/// Converts seconds into days, hours, minutes
+fn days_hours_mins(secs: u32) -> (u32, u32, u32) {
+    let days = secs / 60 / 60 / 24;
+    let hours = secs / 60 / 60 % 24;
+    let minutes = secs / 60 % 60;
+    (days, hours, minutes)
+}
+
+/// Converts seconds into a string in the format: x days, x hours, x mins
+fn uptime_str(secs: u32) -> String {
+    let (days, hours, mins) = days_hours_mins(secs);
+    let mut uptime = String::new();
+
+    if days > 0 {
+        uptime.push_str(&format!("{} days, ", days));
+    }
+    if hours > 0 {
+        uptime.push_str(&format!("{} hours, ", hours));
+    }
+    if mins > 0 {
+        uptime.push_str(&format!("{} mins", mins));
+    }
+    if days == 0 && hours == 0 && mins == 0 {
+        uptime = format!("{} secs", secs);
+    }
+    // In case uptime is exactly some amount of days or hours
+    uptime
+        .trim_end_matches(|c: char| c == ',' || c.is_whitespace())
+        .to_string()
+}
```

Comment:
i'm not sure this is right, maybe we could have a struct for all this?
```
struct Duration {
    days: u32,
    hours: u32,
    mins: u32,
    secs: u32,
}

impl Duration {
    fn from_secs(secs: u32) -> Self {
        ...
    }
}
```

Reply:
and a function to pretty format the duration (taking bool as parameter for short output)